### PR TITLE
Add support for `AddEventListenerOptions` deduplication (lightweight-ly)

### LINKED ISSUE #9: Duplicate detection is broken when using options rather than the capture boolean
The spec says that duplicate detection is based on callback, type and capture only. But your code does not properly implement that when passing an options object as last argument (you test for options objects being the same, which is much less likely).

https://dom.spec.whatwg.org/#add-an-event-listener

### PR DESCRIPTION
**Pull request by bfred-it**
Fixes #9 
Closes #10 

This solution is lighter and not a breaking change

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -25,8 +25,9 @@ function _delegate<TElement extends Element = Element, TEvent extends Event = Ev
 	selector: string,
 	type: delegate.EventType,
 	callback: delegate.DelegateEventHandler<TEvent, TElement>,
-	useCapture?: boolean | AddEventListenerOptions
+	options?: boolean | AddEventListenerOptions
 ): delegate.DelegateSubscription {
+	const optionsHash = JSON.stringify(options);
```
Comment (stof) [inline index.ts]: this is wrong. The DOM spec does not deduplicate based on all options, but only on the `capture` one (and passing `false` or `{ capture: false }` or nothing (as `false` is the default) should all be handled the same for deduplication.

Comment (bfred-it) [reply index.ts]: That doesn't seem to be the case though, these two listeners are deduplicated:

```js
document.addEventListener('click', console.log, {once: true})
document.addEventListener('click', console.log, {once: true})
document.body.click();
```

Comment (stof) [reply index.ts]: these 2 listeners have the same `capture` flag. Try adding them with different values for `once`, and you will see that they are *also* deduplicated.

Comment (stof) [reply index.ts]: what is needed here is to extract the actual capture flag:

```js
const capture = typeof options === 'undefined' ? false: (typeof options === 'object' ? options.capture || false : options)
```

**Pull request merged**